pkg/kubectl/cmd/edit.go

```diff
@@ -257,7 +257,12 @@ func RunEdit(f *cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args
 			}
 			err = schema.ValidateBytes(stripComments(edited))
 			if err != nil {
-				return preservedFile(err, file, errOut)
+				results = editResults{
+					file: file,
+				}
+				containsError = true
+				results.header.reasons = append(results.header.reasons, editReason{head: fmt.Sprintf("The edited file fails validation: %v", err)})
```

Comment:
Can you wrap the error returned from `ValidateBytes` in an `Invalid` api error and use `addError` instead?